### ISSUE TITLE
Screenshot generation is broken

### ISSUE THREAD

**Issue by teolemon** [openfoodfacts/openfoodfacts-androidapp#3094]
1. To activate screenshots buildType in IDE; uncomment def result = "screenshots" and comment the other line otherwise the folder androidTestScreenshots is not recognized as a test folder. 
1bis. connect a physical device to the computer
2. `gradle connectedOffScreenshotsAndroidTest --stacktrace --info -PtestBuildType=screenshots `
3. Bugs
```
[RemoteAndroidTest]: Running CLASSPATH=$(pm path androidx.test.services) app_process / androidx.test.services.shellexecutor.ShellMain am instrument -r -w -e targetInstrumentation openfoodfacts.github.scrachx.openfood.screenshots.test/androidx.test.runner.AndroidJUnitRunner   androidx.test.orchestrator/androidx.test.orchestrator.AndroidTestOrchestrator on Pixel - 10
[InstrumentationResultParser]: test run failed: 'No test results'
Starting 0 tests on Pixel - 10
Tests on Pixel - 10 failed: No test results
[XmlResultReporter]: XML test result file generated at /Users/pierre/Documents/GitHub/openfoodfacts-androidapp/app/build/outputs/androidTest-results/connected/flavors/OFF/TEST-Pixel - 10-app-OFF.xml. Total tests 0, 

com.android.builder.testing.ConnectedDevice > No tests found.[Pixel - 10] FAILED 
No tests found. This usually means that your test classes are not in the form that your test runner expects (e.g. don't inherit from TestCase or lack @Test annotations).
[XmlResultReporter]: XML test result file generated at /Users/pierre/Documents/GitHub/openfoodfacts-androidapp/app/build/outputs/androidTest-results/connected/flavors/OFF/TEST-Pixel - 10-app-OFF.xml. Total tests 1, failure 1, 
DeviceConnector 'Pixel - 10': uninstalling openfoodfacts.github.scrachx.openfood.screenshots.test
DeviceConnector 'Pixel - 10': uninstalling openfoodfacts.github.scrachx.openfood.screenshots
Task :app:connectedOffScreenshotsAndroidTest in app Finished

> Task :app:connectedOffScreenshotsAndroidTest FAILED
:app:connectedOffScreenshotsAndroidTest (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 41.378 secs.

FAILURE: Build failed with an exception.
```

**Comment by teolemon** [comment]
2020-05-08 21:54:10.967 1119-1198/? E/PackageManager: failed to find package openfoodfacts.github.scrachx.openfood.screenshots.test
2020-05-08 21:54:10.967 1119-1198/? E/OverlayManager: Failed to change enabled overlays for openfoodfacts.github.scrachx.openfood.screenshots.test user 0

**Comment by teolemon** [comment]
[22:42:18]: ▸ /sdcard
[22:42:18]: Cleaning screenshots on device
ls: /sdcard/openfoodfacts.github.scrachx.openfood/screengrab: No such file or directory
[22:42:18]: Exit status: 1
ls: /data/data/openfoodfacts.github.scrachx.openfood/app_screengrab: No such file or directory
[22:42:18]: Exit status: 1
ls: /data/data/openfoodfacts.github.scrachx.openfood/screengrab: No such file or directory
[22:42:18]: Exit status: 1
ls: /data/user/0/openfoodfacts.github.scrachx.openfood/files/openfoodfacts.github.scrachx.openfood/screengrab/en-US/images/screenshots: No such file or directory
[22:42:18]: Exit status: 1
ls: /data/data/openfoodfacts.github.scrachx.openfood/files/openfoodfacts.github.scrachx.openfood/screengrab/en-US/images/screenshots: No such file or directory
[22:42:18]: Exit status: 1
[22:42:18]: To not be asked about this value, you can specify it using 'app_apk_path'
[22:42:18]: Select your debug app APK

**Comment by teolemon** [comment]
Security exception: Package openfoodfacts.github.scrachx.openfood has not requested permission android.permission.CHANGE_CONFIGURATION

java.lang.SecurityException: Package openfoodfacts.github.scrachx.openfood has not requested permission android.permission.CHANGE_CONFIGURATION
	at com.android.server.pm.permission.BasePermission.enforceDeclaredUsedAndRuntimeOrDevelopment(BasePermission.java:425)
	at com.android.server.pm.permission.PermissionManagerService.grantRuntimePermission(PermissionManagerService.java:2114)
	at com.android.server.pm.permission.PermissionManagerService.access$900(PermissionManagerService.java:121)
	at com.android.server.pm.permission.PermissionManagerService$PermissionManagerServiceInternalImpl.grantRuntimePermission(PermissionManagerService.java:3018)
	at com.android.server.pm.PackageManagerService.grantRuntimePermission(PackageManagerService.java:5735)
	at com.android.server.pm.PackageManagerShellCommand.runGrantRevokePermission(PackageManagerShellCommand.java:1955)
	at com.android.server.pm.PackageManagerShellCommand.onCommand(PackageManagerShellCommand.java:230)
	at android.os.ShellCommand.exec(ShellCommand.java:104)
	at com.android.server.pm.PackageManagerService.onShellCommand(PackageManagerService.java:21745)
	at android.os.Binder.shellCommand(Binder.java:881)
	at android.os.Binder.onTransact(Binder.java:765)
	at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:4860)
	at com.android.server.pm.PackageManagerService.onTransact(PackageManagerService.java:4015)
	at android.os.Binder.execTransactInternal(Binder.java:1021)
	at android.os.Binder.execTransact(Binder.java:994)
[23:02:09]: Exit status: 255

**Comment by teolemon** [comment]
[23:40:33]: The `aapt` command could not be found, so status bar could not be cleaned. Make sure android_home is configured for screengrab or ANDROID_HOME is set in the environment
[23:40:33]: Running tests for locale: en-US
[23:40:33]: $ /Users/pierre/Library/Android/sdk/platform-tools/adb -s FA69N0303526 shell am instrument --no-window-animation -w \
-e testLocale en_US \
-e endingLocale en_US \
-e appendTimestamp true \
org.openfoodfacts.scanner.test/androidx.test.runner.AndroidJUnitRunner
[23:40:34]: ▸ android.util.AndroidException: INSTRUMENTATION_FAILED: org.openfoodfacts.scanner.test/androidx.test.runner.AndroidJUnitRunner
[23:40:34]: ▸ 	at com.android.commands.am.Instrument.run(Instrument.java:514)
[23:40:34]: ▸ 	at com.android.commands.am.Am.runInstrument(Am.java:196)
[23:40:34]: ▸ 	at com.android.commands.am.Am.onRun(Am.java:80)
[23:40:34]: ▸ 	at com.android.internal.os.BaseCommand.run(BaseCommand.java:56)
[23:40:34]: ▸ 	at com.android.commands.am.Am.main(Am.java:50)
[23:40:34]: ▸ 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
[23:40:34]: ▸ 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:338)
[23:40:34]: ▸ FORTIFY: pthread_mutex_lock called on a destroyed mutex (0x71286f6000)
android.util.AndroidException: INSTRUMENTATION_FAILED: org.openfoodfacts.scanner.test/androidx.test.runner.AndroidJUnitRunner
	at com.android.commands.am.Instrument.run(Instrument.java:514)
	at com.android.commands.am.Am.runInstrument(Am.java:196)
	at com.android.commands.am.Am.onRun(Am.java:80)
	at com.android.internal.os.BaseCommand.run(BaseCommand.java:56)
	at com.android.commands.am.Am.main(Am.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:338)
FORTIFY: pthread_mutex_lock called on a destroyed mutex (0x71286f6000)
[23:40:34]: Exit status: 1

**Comment by teolemon** [comment]
It seems mostly fixed somehow. Closing

**Comment by VaiTon** [comment]
Fixed by c509ca58ac871ffa86a7320bb349e127ed8ab45b
